INDEXING.md

```diff
@@ -4,48 +4,29 @@
+* An object is part of a batch of objects queued for bulk indexing within Argo (deprecated)
 
-In both cases, Argo uses dor-services to:
+To support indexing, dor-services will:
 * load the object from Fedora.
 * create a Solr document representing the object (using its `to_solr` method).
 * save the document to the Solr instance specified in the configuration, possibly committing immediately.
 
-
 ## Argo::Indexer
 
+This is deprecated for `Dor::IndexingService` in `dor-services`.
+
 ### Reusable methods
 
 The `Argo::Indexer` class provides re-usable methods for indexing:
 * `Argo::Indexer.reindex_pid` takes a PID (druid) and reindexes it as per the above explanation (load from Fedora, create Solr doc, save to Solr).  It also takes an optional logger to use instead of the default (which might be useful for something like logging info about a request triggering the reindex from in the Rails app).  Finally, it gives the option to swallow exceptions.  The default is to let them bubble up, but a caller doing bulk indexing might wish to proceed and index everything it can, with the expectation that the logs will be reviewed later for errors (though note that only `StandardError` errors get trapped, all others always propogate).  The outcome of each indexing attempt is logged, with the PID (druid) and status of the attempt.
 * `Argo::Indexer.generate_index_logger` will create a logger instance that writes to the file specified in `Settings.INDEXER.LOG` with leading string obtained by executing the block provided to `generate_index_logger`.
 
-### Example usage by DorController (which backs the `/dor/reindex` endpoint in Argo)
-
-`DorController.index_logger` uses `Argo::Indexer.generate_index_logger` to generate a logger that leads with the request ID, since those logging events will be triggered from a web request:
-```ruby
-  def index_logger
-    @index_logger ||= Argo::Indexer.generate_index_logger { request.uuid }
-  end
-```
-
-`DorController.reindex` calls `Argo::Indexer.reindex_pid` with the logger instantiated by `index_logger`, and asks Solr to commit the added document (since calling code may expect the update to be available immediately).  Since it uses the default behavior of letting errors bubble up, it specifically deals with `ActiveFedora::ObjectNotFoundError`, rendering a 404 response.
-```ruby
-  def reindex
-    solr_doc = Argo::Indexer.reindex_pid params[:pid], index_logger
-    Dor::SearchService.solr.commit
-    render :plain => "Status:ok<br> Solr Document: #{solr_doc.inspect}"
-  rescue ActiveFedora::ObjectNotFoundError # => e
-    render :status => 404, :plain => 'Object does not exist in Fedora.'
-    return
-  end
-```
-
-
 ## Bulk reindexing
 
+This method is deprecated. See the [`dor_indexing_app`](https://github.com/sul-dlss/dor_indexing_app) Wiki for the new methods. The `Argo::PidGatherer`, however, is not migrated so if you need to gather pids directly from Fedora, refer to that code.
+
```

Comment:
We were successful doing a full reindex in ~3 days using sulmq only. It was not a from-scratch index (where we needed to do ordering by object type), just a reindex of all the pids in the Solr index already. We weren't asked to do a from-scratch index, but I think your PidGatherer approach is still valid for that.

I'm not sure about the flakeyness with the message broker approach. dor_indexing_app is much faster and stable now, so it may be more reliable.

The Argo bulk update for reindexing is still supported (the javascript stuff). See https://github.com/sul-dlss/dor_indexing_app/issues/105 for that work.